src/Browscap/Coverage/Processor.php

```diff
@@ -0,0 +1,571 @@
+                        $coverage['statementMap'][] = [
+                            'start' => $deviceDefinitionStart,
+                            'end' => ['line' => $lexer->yylineno + 1, 'column' => strpos($line, '}') + 1],
+                        ];
+                        $coverage['s'][] = array_sum(array_column($locations, 'coverCount'));
+
+                        $coverArray = [];
+
+                        foreach ($locations as $location) {
+                            $coverage['statementMap'][] = [
+                                'start' => $location['start'],
+                                'end' => $location['end'],
+                            ];
+                            $coverage['s'][] = $location['coverCount'];
+                            $coverArray[]    = $location['coverCount'];
+                        }
+
+                        $coverage['b'][] = $coverArray;
+
+                        foreach (array_keys($branch['locations']) as $index) {
+                            unset($branch['locations'][$index]['coverCount']);
+                        }
+
+                        $coverage['branchMap'][] = $branch;
+
+                        $sectionBranches = [];
+                    }
+                    break;
+            }
+        } while ($code !== 1);
+
+        // This re-indexes the arrays to be 1 based instead of 0, which will make them be JSON objects rather
+        // than arrays, which is how they're expected to be in the coverage JSON file
+        $coverage['fnMap']        = array_filter(array_merge([0], $coverage['fnMap']));
+        $coverage['statementMap'] = array_filter(array_merge([0], $coverage['statementMap']));
+        $coverage['branchMap']    = array_filter(array_merge([0], $coverage['branchMap']));
+        array_unshift($coverage['b'], '');
+        unset($coverage['b'][0]);
+        array_unshift($coverage['f'], '');
+        unset($coverage['f'][0]);
+        array_unshift($coverage['s'], '');
+        unset($coverage['s'][0]);
+
+        return $coverage;
+    }
+
+    /**
+     * Groups pattern ids by their filename prefix
+     *
+     * @param  array $ids
+     * @return array
+     */
+    private function groupIdsByFile(array $ids)
+    {
+        $covered = [];
+
+        foreach ($ids as $id) {
+            $file = substr($id, 0, strpos($id, '::'));
+
+            if (!isset($covered[$file])) {
+                $covered[$file] = [];
+            }
+
+            $covered[$file][] = substr($id, strpos($id, '::') + 2);
+        }
+
+        return $covered;
+    }
+
+    /**
+     * Counts number of times given pattern is covered by test patterns
+     *
+     * @param  string $id
+     * @param  array  $covered
+     * @return int
+     */
+    private function getCoverageCount($id, array $covered)
+    {
+        $id                  = str_replace('\/', '/', $id);
+        list($u, $c, $d, $p) = explode('::', $id);
```

Comment:
What are `$u`, `$c`, `$d` and `$p`? They tell me nothing about what they're doing